WPMedia/fusion-news-theme-blocks
# Tmedia 781 use GitHub token over pat

## Description

Use more secure and scalable github token when possible over pat 

## Jira Ticket

- [TMEDIA-781](https://arcpublishing.atlassian.net/browse/TMEDIA-781)
- Will also make pr for 2.0 when ready

## Acceptance Criteria

- Cache works as before. I'm triggering a new cache 
- Publish works without individual's pat. Since it's only committing without pushing, this worked as expected in themes lerna repo https://github.com/WPMedia/themes-lerna-playground/runs/5740584973?check_suite_focus=true#step:6:1

## Test Steps

- Notice working as expected here https://github.com/WPMedia/themes-lerna-playground/runs/5740584973?check_suite_focus=true#step:6:1 
- Same security branching protection 
- See github actor and github token used here in-line. Mike rec'd they go in-line as each action needs that config in scope

### Caching 

- The previous commit updated the package-lock, triggering a reinstall of the cache 
- See successful test and cache https://github.com/WPMedia/arc-themes-blocks/runs/5820354365?check_suite_focus=true
- Created an empty commit that should not create a new cache 
- Tests and caching still pass https://github.com/WPMedia/arc-themes-blocks/runs/5820443182?check_suite_focus=true
- https://github.com/WPMedia/themes-lerna-playground/blob/canary/.github/workflows/publish-canary.yml#L38

## Effect Of Changes

### Before

- Using one person's pat. Which could be revoked at any time even accidentally 

### After

- Using best practices for github and app-sec (cc @manchicken ) 

## Dependencies or Side Effects

- Reminder: Remove personal auth token from repo after this is complete 
- Also removed unused workflow of the stable dist parity
- I fixed a typo in a readme to trigger a block publish as well

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._
